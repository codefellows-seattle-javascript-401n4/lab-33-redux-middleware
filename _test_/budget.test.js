'use strict';


import React from 'react';
import renderer from 'react-test-renderer';
import Jest from 'jest';
import uuid from 'uuid/v1';
import CatForm from '../src/components/catForm.js';

test('test should pass and snapshot has been added', () => {
  let name = 'groceries';
  let budget = 10;
  let createBudget = (state) => {
    expect(CatForm.state().name).toBe(name);
    expect(CatForm.state().budget).toBe(budget);
  };
  const component = renderer.create(
    <CatForm />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('test should fail because change in name of budget', () => {
  const component = renderer.create(
    <CatForm name="groceries"/>
  );
  const other = renderer.create(
    <CatForm name="gas"/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let otherTree = other.toJSON();
  expect(tree).toEqual(otherTree);
});

test('test should pass with unique ids', () => {
  const component = renderer.create(
    <CatForm id={uuid()}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('test should fail because non unique ids', () => {
  const component = renderer.create(
    <CatForm id="1"/>
  );
  const other = renderer.create(
    <CatForm id="1"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  let otherTree = other.toJSON();
  expect(otherTree).toEqual(tree);
});
