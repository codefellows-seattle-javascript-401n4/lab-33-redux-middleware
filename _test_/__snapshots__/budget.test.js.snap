// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should fail because change in name of budget 1`] = `
<div>
  <form
    className="catForm"
    onSubmit={[Function]}
  >
    <label
      id="catFormName"
    >
       Enter Your Budget Category 
    </label>
    <input
      htmlFor="catFormName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="catFormBudget"
    >
       Enter Your Budget Amount 
    </label>
    $
    <input
      htmlFor="catFormBudget"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`test should fail because non unique ids 1`] = `
<div>
  <form
    className="catForm"
    onSubmit={[Function]}
  >
    <label
      id="catFormName"
    >
       Enter Your Budget Category 
    </label>
    <input
      htmlFor="catFormName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="catFormBudget"
    >
       Enter Your Budget Amount 
    </label>
    $
    <input
      htmlFor="catFormBudget"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`test should pass and snapshot has been added 1`] = `
<div>
  <form
    className="catForm"
    onSubmit={[Function]}
  >
    <label
      id="catFormName"
    >
       Enter Your Budget Category 
    </label>
    <input
      htmlFor="catFormName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="catFormBudget"
    >
       Enter Your Budget Amount 
    </label>
    $
    <input
      htmlFor="catFormBudget"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`test should pass with unique ids 1`] = `
<div>
  <form
    className="catForm"
    onSubmit={[Function]}
  >
    <label
      id="catFormName"
    >
       Enter Your Budget Category 
    </label>
    <input
      htmlFor="catFormName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="catFormBudget"
    >
       Enter Your Budget Amount 
    </label>
    $
    <input
      htmlFor="catFormBudget"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
